pytorch_lightning/trainer/trainer.py

```diff
@@ -2475,6 +2476,33 @@ def _active_loop(self) -> Optional[Union[FitLoop, EvaluationLoop, PredictionLoop
     Logging properties
     """
 
+    @property
+    def logger(self) -> Optional[LightningLoggerBase]:
+        return self._logger
+
+    @logger.setter
+    def logger(self, new_logger: Optional[LightningLoggerBase]) -> None:
+        self._logger = new_logger
+        if not new_logger:
+            self._loggers = []
+        elif isinstance(new_logger, LoggerCollection):
+            self._loggers = list(new_logger)
+        else:
+            self._loggers = [new_logger]
+
+    @property
+    def loggers(self) -> List[LightningLoggerBase]:
+        return self._loggers
+
+    @loggers.setter
+    def loggers(self, new_loggers: Optional[Iterable[LightningLoggerBase]]) -> None:
```

Comment:
I still feel like this should take in a `List[LightningLoggerBase]]` since the property `loggers` is of that type. What is the end state we want when the deprecation cycle is complete? A property `loggers` which is of type list and can only be set to a type list.